src/vcpkg/triplet.cpp

```diff
@@ -86,24 +86,26 @@ namespace vcpkg
             default: return Triplet::from_canonical_name("x86-windows");
         }
 #elif defined(__APPLE__)
-        return Triplet::from_canonical_name("x64-osx");
+        switch (host_proc)
```

Comment:
If we're switching this to be runtime, I think it would be cleaner to go all-out and assemble:
```cpp
#if $platform
StringLiteral platform = "$platform"
#endif
return Triplet::from_canonical_name(Strings::concat(to_zstring_view(host_proc), "-", platform));
```

Reply:
hmm, fair enough.

Reply:
Even if the triplets don't exist now, I don't think picking an obviously wrong one is the right behavior.